### PR TITLE
sampling/test: Cleanup processor stop, 1ms timeout

### PR DESCRIPTION
**Pull request by marclop**
## Motivation/summary

Updates the sampling tests to only wait up to `1ms` for the processor to stop on cleanup. Nearly halving the unit tests time for that package:

```
ok  github.com/elastic/apm-server/x-pack/apm-server/sampling	85.742s
ok  github.com/elastic/apm-server/x-pack/apm-server/sampling	44.517s
```

## Checklist

~- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~
~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

N/A

## Related issues

N/A

### CODE REVIEW

File: x-pack/apm-server/sampling/processor_test.go [hunk 1]
```diff
@@ -905,3 +905,11 @@ func waitFileModified(tb testing.TB, filename string, after time.Time) ([]byte,
 		}
 	}
 }
+
+func cleanupProcessor(t *testing.T, processor *sampling.Processor) {
+	t.Cleanup(func() {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond)
+		defer cancel()
+		processor.Stop(ctx)
```
Comment (axw) [inline x-pack/apm-server/sampling/processor_test.go:913]: Is there even a point in waiting 1ms? Should we use WithCancel and immediately cancel the context?

File: x-pack/apm-server/sampling/processor_test.go [hunk 2]
```diff
@@ -905,3 +905,11 @@ func waitFileModified(tb testing.TB, filename string, after time.Time) ([]byte,
 		}
 	}
 }
+
+func cleanupProcessor(t *testing.T, processor *sampling.Processor) {
```
Comment (axw) [inline x-pack/apm-server/sampling/processor_test.go:909]: WDYT about a new function which both creates and starts the processor, and adds a cleanup to stop it with immediate cancellation?

e.g.

```go
func newProcessor(t testing.TB, cfg sampling.Config) *sampling.Processor
```

**Review by axw — COMMENTED** [review]
Thanks for looking into this @marclop!

File: x-pack/apm-server/sampling/processor_test.go (hunk 1 above)
Comment (marclop) [reply x-pack/apm-server/sampling/processor_test.go:913]: I did that initially but got a panic, so I decided to use 1ms which shouldn't really affect the test runtime anyway. I haven't looked into what exactly triggers this panic, though.

```
panic: Transaction has already been discarded

goroutine 102526 [running]:
github.com/dgraph-io/badger/v2.(*Txn).NewIterator(0x1a8317f4cb1?, {0x1, 0x64, 0x0, 0x0, {0x140000438c0, 0x25, 0x30}, 0x0, 0x0})
	/Users/marclop/go/pkg/mod/github.com/dgraph-io/badger/v2@v2.2007.3-0.20201012072640-f5a7e0a1c83b/iterator.go:460 +0x4b4
github.com/elastic/apm-server/x-pack/apm-server/sampling/eventstorage.(*ReadWriter).ReadTraceEvents(0x14000f17440, {0x14000043890?, 0x24}, 0x14000650048)
	/Users/marclop/repos/elastic/apm-server/x-pack/apm-server/sampling/eventstorage/storage.go:220 +0x170
github.com/elastic/apm-server/x-pack/apm-server/sampling/eventstorage.(*lockedReadWriter).ReadTraceEvents(0x14000ce1530?, {0x14000043890?, 0x18?}, 0x105cdf500?)
	/Users/marclop/repos/elastic/apm-server/x-pack/apm-server/sampling/eventstorage/sharded.go:110 +0xd4
github.com/elastic/apm-server/x-pack/apm-server/sampling/eventstorage.(*ShardedReadWriter).ReadTraceEvents(0x14001080b58?, {0x14000043890, 0x24}, 0x28?)
	/Users/marclop/repos/elastic/apm-server/x-pack/apm-server/sampling/eventstorage/sharded.go:56 +0x3c
github.com/elastic/apm-server/x-pack/apm-server/sampling.(*wrappedRW).ReadTraceEvents(...)
	/Users/marclop/repos/elastic/apm-server/x-pack/apm-server/sampling/processor.go:586
github.com/elastic/apm-server/x-pack/apm-server/sampling.(*Processor).Run.func7()
	/Users/marclop/repos/elastic/apm-server/x-pack/apm-server/sampling/processor.go:465 +0x1e8
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/Users/marclop/go/pkg/mod/golang.org/x/sync@v0.0.0-20220819030929-7fc1605a5dde/errgroup/errgroup.go:75 +0x60
created by golang.org/x/sync/errgroup.(*Group).Go
	/Users/marclop/go/pkg/mod/golang.org/x/sync@v0.0.0-20220819030929-7fc1605a5dde/errgroup/errgroup.go:72 +0xa8
```

**Comment by axw** [comment]
I've put up an alternative fix here: https://github.com/elastic/apm-server/pull/9838

The issue was that we were unconditionally waiting for the grace period (5 seconds) on shutdown.

**Pull request closed without merging**